spring-integration-file/src/test/java/org/springframework/integration/file/remote/StreamingInboundTests.java

```diff
@@ -0,0 +1,223 @@
+		assertEquals("qux", received.getPayload());
+		assertEquals("/foo", received.getHeaders().get(FileHeaders.REMOTE_DIRECTORY));
+		assertEquals("bar", received.getHeaders().get(FileHeaders.REMOTE_FILE));
+		assertNull(out.receive(0));
+
+		verify(new IntegrationMessageHeaderAccessor(receivedStream).getCloseableResource()).close();
+	}
+
+	public static class Streamer extends AbstractRemoteFileStreamingMessageSource<String> {
+
+		protected Streamer(RemoteFileTemplate<String> template, Comparator<AbstractFileInfo<String>> comparator) {
+			super(template, comparator);
+		}
+
+		@Override
+		public String getComponentType() {
+			return "Streamer";
+		}
+
+		@Override
+		protected List<AbstractFileInfo<String>> asFileInfoList(Collection<String> files) {
+			List<AbstractFileInfo<String>> infos = new ArrayList<AbstractFileInfo<String>>();
+			for (String file : files) {
+				infos.add(new StringFileInfo(file));
+			}
+			return infos;
+		}
+
+	}
+
+	public static class StringFileInfo extends AbstractFileInfo<String> {
+
+		private final String name;
+
+		private StringFileInfo(String name) {
+			this.name = name;
+		}
+
+		@Override
+		public boolean isDirectory() {
+			return false;
+		}
+
+		@Override
+		public boolean isLink() {
+			return false;
+		}
+
+		@Override
+		public long getSize() {
+			return 0;
+		}
+
+		@Override
+		public long getModified() {
+			return 0;
+		}
+
+		@Override
+		public String getFilename() {
+			return this.name.substring(this.name.lastIndexOf("/") + 1);
+		}
+
+		@Override
+		public String getPermissions() {
+			return null;
+		}
+
+		@Override
+		public String getFileInfo() {
+			return null;
+		}
+
+	}
+
+	public static class StringRemoteFileTemplate extends RemoteFileTemplate<String> {
+
+		public StringRemoteFileTemplate(SessionFactory<String> sessionFactory) {
+			super(sessionFactory);
+		}
+
+	}
+
+	public static class StringSessionFactory implements SessionFactory<String> {
+
+		@SuppressWarnings("unchecked")
+		@Override
+		public Session<String> getSession() {
+			try {
+				Session<String> session = mock(Session.class);
+				willReturn(new String[] { "/foo/foo", "/foo/bar" }).given(session).list("/foo");
+				ByteArrayInputStream foo = new ByteArrayInputStream("foo\nbar".getBytes());
+				ByteArrayInputStream bar = new ByteArrayInputStream("baz\nqux".getBytes());
+				willReturn(foo).given(session).readRaw("/foo/foo");
+				willReturn(bar).given(session).readRaw("/foo/bar");
+
+				willReturn(new String[] { "/bar/foo", "/bar/bar" }).given(session).list("/bar");
+				ByteArrayInputStream foo2 = new ByteArrayInputStream("foo\r\nbar".getBytes());
+				ByteArrayInputStream bar2 = new ByteArrayInputStream("baz\r\nqux".getBytes());
+				willReturn(foo2).given(session).readRaw("/bar/foo");
+				willReturn(bar2).given(session).readRaw("/bar/bar");
+
+				given(session.finalizeRaw()).willReturn(true);
+				return session;
+			}
+			catch (Exception e) {
+				fail("failed to mock session");
```

Comment:
Why rethrow some `RuntimeException` isn't enough?
Won't this `fail()` hide the real issue, if someone will modify the mocking wrong way?